internal/app/cache/cache.go

```diff
@@ -85,8 +85,23 @@ func (c *cache) Fetch(key string) (*Response, error) {
 	}
 	// Lazy eviction based on TTL occurs here. Fetch does not increase the lifespan of the key.
 	if resource.isExpired(time.Now()) {
-		c.cache.Remove(key)
-		return nil, nil
+		c.cacheMu.Lock()
+		defer c.cacheMu.Unlock()
+		value, found = c.cache.Get(key)
+		if !found {
+			// The entry was already evicted.
+			return nil, nil
+		}
+		resource, ok = value.(Resource)
+		if !ok {
+			return nil, fmt.Errorf("unable to cast cache value to type resource for key: %s", key)
+		}
+		if resource.isExpired(time.Now()) {
```

Comment:
We're checking `isExpired` again in the case of a recent concurrent `set` right?

Reply:
Would just document this a bit, but approach makes sense to me.

Reply:
yep, added some comments